charts/newrelic-infrastructure-beta/templates/control-plane/daemonset.yaml

```diff
@@ -0,0 +1,211 @@
+{{- if and (.Values.controlPlane.enabled) (not (include "newrelic.fargate" .)) }}
+apiVersion: apps/v1
+kind: {{ .Values.controlPlane.kind }}
+metadata:
+  namespace: {{ .Release.Namespace }}
+  labels: {{ include "newrelic.labels" . | nindent 4 }}
+  name: {{ template "newrelic.fullname" . }}-control-plane
+  {{- with .Values.controlPlane.annotations }}
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- with .Values.updateStrategy }}
+  updateStrategy: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "newrelic.name" . }}-control-plane
+      app.kubernetes.io/instance: {{ .Release.Name }}-control-plane
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        app.kubernetes.io/name: {{ include "newrelic.name" . }}-control-plane
+        app.kubernetes.io/instance: {{ .Release.Name }}-control-plane
+        {{- with .Values.podLabels }}
+          {{- toYaml . | nindent 8 }}
+        {{- end }}
+        mode: {{ include "newrelic.mode" . }}
+    spec:
+      {{- with .Values.images.pullSecrets }}
+      imagePullSecrets:
+          {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ template "newrelic.serviceAccountName" . }}-control-plane
+      {{- if .Values.privileged }}
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      {{- end }}
+      containers:
+        - name: control-plane
+          image: {{ .Values.images.scraper.repository }}:{{ .Values.images.scraper.tag | default .Chart.AppVersion }}
+          imagePullPolicy: {{ .Values.images.scraper.pullPolicy }}
+          securityContext:
+            runAsUser: {{ .Values.runAsUser }}
+            runAsGroup: {{ .Values.runAsGroup }}
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: true
+          env:
+            - name: "NRI_KUBERNETES_HTTPSERVERPORT"
+              value: "8001"
```

Comment:
i thought that we were using the config `HTTPServerPort` to configure both , the agent and the integration to hit the agent endpoint, otherwise this config might be changed but this will fail since the agent will be listening in other port